providers/databricks/src/airflow/providers/databricks/plugins/databricks_workflow.py

```diff
@@ -148,7 +148,10 @@ def _get_dagrun(dag, run_id: str, session: Session) -> DagRun:
         if not session:
             raise AirflowException("Session not provided.")
 
-        return session.query(DagRun).filter(DagRun.dag_id == dag.dag_id, DagRun.run_id == run_id).first()
+        dag_run = session.query(DagRun).filter(DagRun.dag_id == dag.dag_id, DagRun.run_id == run_id).first()
+        if not dag_run:
+            raise AirflowException("DagRun not found.")
```

Comment:
This combination should be unique so let’s just use `one()` here and let SQLAlchemy report any exceptions for us.